packages/kuma-gui/src/app/services/views/ServiceDetailView.vue

```diff
@@ -18,53 +18,57 @@
       <div
         class="stack"
       >
-        <KCard>
-          <DataLoader
-            :src="uri(sources, `/meshes/:mesh/service-insights/:name`, {
-              mesh: route.params.mesh,
-              name: route.params.service,
-            })"
-            v-slot="{ data }"
+        <DataLoader
+          :src="uri(sources, `/meshes/:mesh/service-insights/:name`, {
+            mesh: route.params.mesh,
+            name: route.params.service,
+          })"
+          v-slot="{ data }"
+        >
+          <XAboutSection
+            :title="t('services.internal-service.about.title')"
+            :created="data.creationTime ? t('common.formats.datetime', { value: Date.parse(data.creationTime) }) : undefined"
+            :modified="data.modificationTime ? t('common.formats.datetime', { value: Date.parse(data.modificationTime)}) : undefined"
           >
-            <div class="columns">
-              <DefinitionCard>
-                <template #title>
-                  {{ t('http.api.property.status') }}
-                </template>
-
-                <template #body>
-                  <StatusBadge :status="data.status" />
-                </template>
-              </DefinitionCard>
+            <DefinitionCard layout="horizontal">
+              <template #title>
+                {{ t('http.api.property.status') }}:
+              </template>
 
-              <DefinitionCard>
-                <template #title>
-                  {{ t('http.api.property.address') }}
-                </template>
+              <template #body>
+                <StatusBadge :status="data.status" />
+              </template>
+            </DefinitionCard>
 
-                <template #body>
-                  <TextWithCopyButton
-                    v-if="data.addressPort"
-                    :text="data.addressPort"
-                  />
+            <DefinitionCard layout="horizontal">
+              <template #title>
+                {{ t('http.api.property.address') }}:
+              </template>
 
-                  <template v-else>
-                    {{ t('common.detail.none') }}
-                  </template>
+              <template #body>
+                <template v-if="data.addressPort">
+                  <XBadge appearance="decorative">
+                    <XCopyButton :text="data.addressPort" />
+                  </XBadge>
```

Comment:
I believe these shouldn't use `XBadge` at all. They should just use `XCopyButton` in order to achieve what we are trying to do here.

It would be good to add a `:variant="badge"` to `XCopyButton`. It already has a `format` but I think that refers to the format of the text you are about to copy, not the 'variant'/look and feel.

Reply:
I also completely agree with what you are saying i.e:

>  I'd prefer the composable strategy, which separates concerns and allows more flexibility.

Thing is, if someone changes KCopy's to use a different type of badge (not appearance="decorative") then we may hit issues because our "AppAboutSection" will still use appearance="decorative" i.e. this purple look, while everywhere else it will use the "updated upstream look" maybe I dunno blue.

This my main concern, as from what I understand it goes against the reasonings behind this PR. I'm not sure if flexibility is what we are trying to achieve here.

Honestly I don't know what to suggest here myself as yet, still thinking.

Reply:
Aah 😅 I've checked and realized that `KCopy` has a prop `badge` already. I thought this was a custom implementation further downstream.
Now after I realized that, I think I'd go for the `:variant` to stay consistent with upstream changes.

Reply:
I've added the `variant` prop for the `badge` appearance and applied it to the places in `XAboutSection`.